### DISCUSSION TITLE
Erro ao Armazenar JWT TOKEN de uma API externa

### DISCUSSION THREAD

**Discussion by Gondlir** [balta-io/7187#83] [Perguntas e Respostas] [answered]
Eu tenho uma API desenvolvida em C# e .NET 6 que está hospedada no Azure e que ao fazer requisição de autenticação retorna um ActionResult com um Token e RefreshToken
Só que seguindo as aulas do 7187, a função que salva o TOKEN não está salvando de maneira apropriada e eu não entendo o porquê...
No meu localStorage ele fica tomo 'token' undefined
o meu security.utils.ts está o mesmo que do vídeo mas ele não ta armazenando o TOKEN de maneira apropriada 

![image](https://user-images.githubusercontent.com/32470349/185793132-d1fdf6ca-8259-4c2e-a91c-576368f8d3c0.png)
![image](https://user-images.githubusercontent.com/32470349/185793140-b63215a0-dbd2-42cc-8f44-0244448141d4.png)

**Comment by Gondlir** [comment]
Outro detalhe, eu não sei o pq toda vez que clico o botão de login 'uma vez' ele faz **dois request** ao invés de um, nisso em um ele retorna 204'NO CONTENT' e no outro ele autentica normal...

**Comment by andrebaltieri** [comment, marked as answer]
@Gondlir é difícil saber, podem ser muitas coisas... precisa debugar seu JS e ver por que ele está enviando UNDEFINED para o Local Storage se o token vem correto do backend.

Em relação a executar duas vezes, pode ser que esteja chamando a mesma função duas vezes ou recarregando a página, novamente, sem um debug mais profundo é difícil dizer.

Se possível, inspeciona melhor, com mais detalhes o código e retorna aqui que conseguimos ajudamos melhor 💜
